protocol/contracts/templegold/SpiceAuction.sol

```diff
@@ -0,0 +1,494 @@
+    mapping(uint256 auctionId => SpiceAuctionConfig config) public auctionConfigs;
+    /// @notice Keep track of claimed amounts per auction token. This helps calculate amount of auction tokens for next epoch.
+    mapping(address token => uint256 amount) private _claimedAuctionTokens;
+    /// @notice Keep track of total allocation per auction token
+    mapping(address token => uint256 amount) private _totalAuctionTokenAllocation;
+    /// @notice Keep track of redeemed and notified epochs
+    mapping(uint256 epochId => bool redeemed) public override redeemedEpochs;
+
+    constructor(
+        address _templeGold,
+        address _spiceToken,
+        address _daoExecutor,
+        address _operator,
+        uint32 _arbEid,
+        uint32 mintChainId_,
+        string memory _name
+    ) {
+        spiceToken = _spiceToken;
+        daoExecutor = _daoExecutor;
+        operator = _operator;
+        templeGold = _templeGold;
+        _arbitrumOneLzEid = _arbEid;
+        _mintChainId = mintChainId_;
+        name = _name;
+        _deployTimestamp = block.timestamp;
+        lzReceiveExecutorGas = 85_412;
+    }
+
+    /**
+     * @notice Set lzReceive gas used by executor
+     * @param _gas Redemption notifier
+     */
+    function setLzReceiveExecutorGas(uint32 _gas) external override onlyOperatorOrDaoExecutor {
+        if (_gas == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        lzReceiveExecutorGas = _gas;
+        emit LzReceiveExecutorGasSet(_gas);
+    }
+
+    /**
+     * @notice Set operator
+     * @param _operator operator to set
+     */
+    function setOperator(address _operator) external override onlyDAOExecutor {
+        if (_operator == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        operator = _operator;
+        emit OperatorSet(_operator);
+    }
+
+    /**
+     * @notice Set DAO executor for DAO actions
+     * @param _daoExecutor New dao executor
+     */
+    function setDaoExecutor(address _daoExecutor) external onlyDAOExecutor {
+        if (_daoExecutor == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        daoExecutor = _daoExecutor;
+        emit DaoExecutorSet(_daoExecutor);
+    }
+
+    /**
+     * @notice Set config for an epoch. This enables dynamic and multiple auctions especially for vested scenarios
+     * @dev Must be set before epoch auction starts
+     * @param _config Config to set
+     */
+    function setAuctionConfig(SpiceAuctionConfig calldata _config) external onlyDAOExecutor {
+        /// @dev epoch Id is only updated when auction starts. 
+        /// @dev cannot set config for past or ongoing auction
+        uint256 currentEpochIdCache = _currentEpochId;
+        if (currentEpochIdCache > 0) {
+            EpochInfo storage info = epochs[currentEpochIdCache];
+            /// Cannot set config for ongoing auction
+            if (info.isActive()) { revert InvalidConfigOperation(); }
```

Comment:
Why is this requirement necessary?
The new `_config` is set for the next epoch, so the current epoch cannot be modified with this function. 

This function could be called at any time because it only applies to the **next** epoch. And as soon as `startAuction()` is called, the `currentEpoch` is increased, so this function would only modify the epoch after this as well. 

Does that make sense or am I missing something?

The current requirement only let's you modify configs in the periods in between auctions, which leaves you less room for reconfiguration. But if you want to leave as it is I would also understand it